### ISSUE TITLE
"react/jsx-curly-spacing": [ 2, "always", { "spacing": {  "objectLiterals": "never" }}] errors on node context

### ISSUE THREAD

**Issue by MoOx** [yannickcr/eslint-plugin-react#671]
I added spacing/objectLiterals config for jsx-curly-spacing and I think I am getting an issue.

```js
  render() {
    return (
      <div
        onClick={ this.handleClick }
        style={{ color: "blue" }} // From doc, it should affect only this part right?
      >
        { // <- There should be no newline after '{'
          !this.state.clicked
          ? "Hi"
          : "Hey"
        } // <- There should be no newline before '{'
      </div>
    )
  }
```

The doc only show an example in props, not sure why it's yelling for nodes.

**Comment by ljharb** [comment]
Yes, you're correct. If that's erroring out, it's a bug.
